lib/graph3d/Settings.js

```diff
@@ -467,6 +469,60 @@ function setDataColor(dataColor, dst) {
   }
 }
 
+/**
+ * Converts an object that accepts a HUE keyword. This is converted to a certain amount of hex color stops. At which point:
+ * the HTML hex color codes is converted into an RGB color object.
+ * 
+ * @param {Object | Array<string>} surfaceColors Either an object that describes the HUE, or an array of HTML hex color codes
+ * @param {Object} dst 
+ */
+function setSurfaceColor(surfaceColors, dst) {
+  if(surfaceColors === undefined) {
+    return;    // Nothing to do
+  }
+
+  if (dst.surfaceColors === undefined) {
+    dst.surfaceColors = {};
+  }
+
+  let rgbColors = [];
+
+  if(Array.isArray(surfaceColors)) {
+    rgbColors = surfaceColors.map(util.hexToRGB);
+  } else if (typeof surfaceColors === 'object') {
+    if(surfaceColors.hue === undefined) {
+      throw new Error('Unsupported type of surfaceColors');
+    }
+
+    const startHue = Math.min(surfaceColors.hue.start, surfaceColors.hue.end);
+    const endHue = Math.max(surfaceColors.hue.start, surfaceColors.hue.end);
+    const saturation = surfaceColors.hue.saturation;
+    const brightness = surfaceColors.hue.brightness;
+    const stops = surfaceColors.hue.colorStops;
+    const hueStepSize = (endHue - startHue) / stops;
+
+    let currentHue = startHue;
+    while (currentHue < endHue) {
+      rgbColors.push(
+        util.HSVToRGB(
+          (currentHue % 360)/360, 
+          saturation/100, 
+          brightness/100
+        )
```

Comment:
If you really don't want to allow negative hues (may I know why?) then you have to take care of them somehow. The way you have it it just glitches for negative hues.

Why not `(360 + (currentHue % 360))` as I suggested instead of `(currentHue % 360)`? Were there any issues with it?

Reply:
Yes if we do `(360 + (currentHue % 360))/360` then the range becomes `1-2` instead of `0-1`